### PR TITLE
Issue Id #211147 feat: Create a Hanging man game in Test-Rig-0.3

### PR DESCRIPTION
**Pull request by ajinkyapandetekdi**
## Summary by CodeRabbit

- **New Features**
	- Expanded form options with additional radio button selections.
	- Introduced a Hangman game component, complete with guessing words, hints, and displaying outcomes.
	- Enhanced user interaction with the addition of splash screens and improved game controls.
- **Enhancements**
	- Improved toast message formatting for better user feedback.
	- Enhanced telemetry service with additional parameters for richer event tracking.
	- Refined audio control and error handling in interactive components.
- **Style**
	- Added styling for the Hangman game interface, ensuring a visually appealing user experience.
- **Refactor**
	- Restructured imports and adjusted state management for clarity and efficiency.
	- Refined API call parameters and error handling for robustness.
- **Bug Fixes**
	- Adjusted rendering logic and functionality in interactive stories to ensure smooth user experiences.

### CODE REVIEW

File: src/config/ConfigForm.jsx [hunk 1]
```diff
@@ -22,38 +34,41 @@ const ConfigForm = () => {
...
+                <FormLabel>{item.title}</FormLabel>
+                <RadioGroup
+                  onChange={value => handleInputChange(index, value)}
+                  value={criteria[index]?.template || ''}
+                >
+                  <HStack>
+                    {' '}
+                    <Radio value="spell-and-check">spell&check</Radio>
+                    <Radio value="simple">Simple</Radio>
+                    <Radio value="hangman-game">hangman</Radio>
```
Comment (gouravmore) [inline src/config/ConfigForm.jsx:67]: already selected config for hangman-game will not show here

File: src/pages/Test-Rig-Two/Story/Story.jsx [hunk 2]
```diff
@@ -340,13 +356,16 @@ const Story = ({ forceRerender, setForceRerender }) => {
   }, [temp_audio]);
 
   const playAudio = () => {
+    interactCall('playAudio', '', 'DT', 'play');
```
Comment (gouravmore) [inline src/pages/Test-Rig-Two/Story/Story.jsx:359]: content id also should log

File: src/pages/Test-Rig-Two/Story/Story.jsx [hunk 3]
```diff
@@ -355,20 +374,28 @@ const Story = ({ forceRerender, setForceRerender }) => {
...
+        '',
+        { err: 'Audio is not available', errtype: 'CONTENT' },
+        'ET',
+        contentId
+      );
     });
   };
 
   const pauseAudio = () => {
+    interactCall('pauseAudio', '', 'DT', 'pause');
```
Comment (gouravmore) [inline src/pages/Test-Rig-Two/Story/Story.jsx:394]: which template is playing audio that should also log

File: src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+          bg={gameWon && letter !== '_' ? 'green.100' : 'transparent'}
+        >
+          {letter}
+        </Text>
+      ))}
+    </HStack>
+  );
+
+  const renderAlphabet = () => {
+    const alphabetRowsEnglish = ['qweiop'];
```
Comment (gouravmore) [inline src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx:154]: why this is hardcoded

File: src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+                    </div>
+                  </Flex>
+                </HStack>
+              </div>
+            )}
+            {!(gameLost || gameWon) && (
+              <Flex
+                onClick={() => {
+                  provideHint();
+                  interactCall('provideHint', '', 'DT', 'hint');
```
Comment (gouravmore) [inline src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx:339]: provideHint for what?

File: src/services/telementryService.js [hunk 6]
```diff
@@ -66,46 +65,54 @@ export const initialize = ({ context, config, metadata }) => {
...
+      mode: 'play',
+      pageid: url,
+      summary: data?.summary || {},
+      duration: data?.duration || '000',
+    },
+  });
 };
 
-export const interact = (telemetryMode) => {
+// export const interact = (telemetryMode) => {
```
Comment (gouravmore) [inline src/services/telementryService.js:102]: what this is commented

File: src/services/telementryService.js [hunk 7]
```diff
@@ -66,46 +65,54 @@ export const initialize = ({ context, config, metadata }) => {
...
+//       edata: { type: 'TOUCH', subtype: '', pageid: url },
+//     });
+//   }
+// };
+
+export const interact = (id, url, telemetryMode, currentPage) => {
   if (checkTelemetryMode(telemetryMode)) {
     CsTelemetryModule.instance.telemetryService.raiseInteractTelemetry({
       options: getEventOptions(),
-      edata: { type: 'TOUCH', subtype: '', pageid: url },
```
Comment (gouravmore) [inline src/services/telementryService.js:108]: subtype removed?

File: src/pages/Test-Rig-Two/Story/Story.jsx [hunk 8]
```diff
@@ -89,75 +91,78 @@ const Story = ({ forceRerender, setForceRerender }) => {
...
-                );
-            } else {
-                return <span key={index}>{word+ ' '}</span>;
-            }
-        });
-        return highlightedSentence;
-      }
-}
+    if (type == 'char' || type == 'word') {
+      // type=='char' || type =='word'
```
[the remark this answers is not in the document]
Comment (ajinkyapandetekdi) [reply src/pages/Test-Rig-Two/Story/Story.jsx:100]: removed

File: src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx [hunk 9]
```diff
@@ -0,0 +1,392 @@
...
+  }, [words]);
+
+  useEffect(() => {
+    resetGame();
+  }, []);
+
+  const resetGame = () => {
+    const randomWord = words[currentWordIndex];
+    setWord(randomWord?.toLowerCase());
+    // const guessWordSplit = randomWord?.split('');
```
Comment (gouravmore) [inline src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx:64]: remove comment if not required

File: src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx (hunk 4 above)
Comment (gouravmore) [reply src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx:157]: random char should show here for keyboard

File: src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx [hunk 10]
```diff
@@ -0,0 +1,392 @@
...
+        );
+      } else {
+        setIncorrectGuesses(prevIncorrectGuesses => prevIncorrectGuesses + 1);
+      }
+    }
+  };
+
+  const handleNextWord = () => {
+    setCurrentWordIndex(prevIndex => prevIndex + 1);
+    // setCurrentLine(prevIndex => prevIndex + 1);
```
[the remark this answers is not in the document]
Comment (ajinkyapandetekdi) [reply src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx]: removed

File: src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx (hunk 4 above)
Comment (ajinkyapandetekdi) [reply src/pages/Test-Rig-Two/Story/hangman-game/HangmanGame.jsx]: This is also Resolved

**Comment by gouravmore** [comment]
@coderabbitai review

**Pull request merged**